Sample crashes on launch

Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
        at android.content.ContextWrapper.getSystemService(ContextWrapper.java:728)
        at android.arch.persistence.room.RoomDatabase$JournalMode.resolve(RoomDatabase.java:388)
        at android.arch.persistence.room.RoomDatabase$Builder.build(RoomDatabase.java:628)
        at com.nytimes.android.sample.SampleRoomStore.<init>(SampleRoomStore.kt:41)
        at com.nytimes.android.sample.SampleApp.<init>(SampleApp.kt:34)

--- a/app/src/main/java/com/nytimes/android/sample/SampleApp.kt
+++ b/app/src/main/java/com/nytimes/android/sample/SampleApp.kt
@@ -30,11 +30,12 @@ class SampleApp : Application() {
     var nonPersistedStore: Store<RedditData, BarCode>? = null
     var  persistedStore: Store<RedditData, BarCode>? =null
     private var persister: Persister<BufferedSource, BarCode>? =null
-    private val sampleRoomStore=SampleRoomStore(this)
+    lateinit var sampleRoomStore:SampleRoomStore
 
     override fun onCreate() {
         super.onCreate()
         appContext = this
+        sampleRoomStore = SampleRoomStore(this)
         initPersister();
         nonPersistedStore = provideRedditStore();
         persistedStore=providePersistedRedditStore();

